perl/toolbox/json.pm

```diff
@@ -16,49 +16,53 @@ our @EXPORT = qw(put_json_file get_json_file open_write_text_file open_read_text
 use strict;
 use warnings;
 
-
 sub open_write_text_file {
     my $filename = shift;
     chomp $filename;
     if (! defined $filename) {
-        printf "open_write_text_file(): filename was not defined\n";
-        return;
+        # filename not defined
+        return 1;
     }
-    if (! $filename =~ /\.xz/) { # Always default to compression when writing
+    if ($filename =~ /\.xz$/) {
+        debug_log(sprintf "file [%s] already named for compression\n", $filename);
```

Comment:
should we prefix the `debug_log` messages with the routine name?  otherwise users of the library may not understand where the message is coming from, especially when the library calls itself and the routing stack gets deep.  for example:

```
debug_log(sprintf "open_write_text_file: file [%s] already named for compression\n", $filename);
```

there may even bee a special string to use so it is done automatically...maybe like this:

```
debug_log(sprintf "%s: file [%s] already named for compression\n", __SUB__, $filename);
```

Reply:
these are in the next push